R/module_nested_tabs.R

```diff
@@ -183,11 +188,40 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, reporter) {
   modules$server_args <- teal.transform::resolve_delayed(modules$server_args, datasets)
   is_module_server <- isTRUE("id" %in% names(formals(modules$server)))
 
-  args <- c(list(id = id, datasets = datasets), modules$server_args)
+  args <- c(list(id = id), modules$server_args)
   if (is_reporter_used(modules)) {
     args <- c(args, list(reporter = reporter))
   }
 
+  is_datasets_used <- isTRUE("datasets" %in% names(formals(modules$server)))
+  if (is_datasets_used) {
+    args <- c(args, datasets = datasets)
+  }
+
+  is_data_used <- isTRUE("data" %in% names(formals(modules$server)))
+  if (is_data_used) {
+    datanames <- if (identical("all", modules$filter)) datasets$datanames() else modules$filter
+
+    # list of reactive filtered data
+    data <- sapply(
+      datanames,
+      simplify = FALSE,
+      function(x) {
+        #todo: need to include metadata, keys, datalabel
+        reactive(datasets$get_data(x, filtered = TRUE))
+      }
+    )
+    #names(data) <- paste0(names(data), "_FILTERED") # these datasets are filtered
```

Comment:
I'd like also to highlight this. `data` contains "filtered data" which is supported by `expr` where `ADSL_FILTERED <- filter(ADSL, ...)`. So, we have little inconsistency here as `data` is named by "generic" dataname (ADSL, ADTTE, ...).

I was thinking that teal.slice should execute expr in such way that `ADSL <- filter(ADSL, <expr>)` to have only one variant.